drivers/bluetooth/hci/ipc.c

```diff
@@ -265,19 +266,30 @@ static int bt_ipc_send(const struct device *dev, struct net_buf *buf)
 		break;
 	default:
 		LOG_ERR("Unknown type %u", bt_buf_get_type(buf));
+		err = -ENOMSG;
 		goto done;
 	}
 	net_buf_push_u8(buf, pkt_indicator);
 
 	LOG_HEXDUMP_DBG(buf->data, buf->len, "Final HCI buffer:");
-	err = ipc_service_send(&data->hci_ept, buf->data, buf->len);
+
+	do {
+		err = ipc_service_send(&data->hci_ept, buf->data, buf->len);
+		if (err < 0) {
+			retries++;
+			k_msleep(CONFIG_BT_HCI_IPC_SEND_RETRY_DELAY_MS);
```

Comment:
@dchat-nordic , can you tell which context will be blocked?

Reply:
I think we should use sleep that will allow us yield. Maybe it should use busy wait if number of us is low and such sleep is not available. Adding another config option would be an overhead but it could be useful in this case.

Reply:
AFAIK k_msleep will put the thread to sleep and reschedule. Do you mean to use k_usleep, or I missed the point of the question?

Reply:
> Do you mean to use k_usleep

Yes, as this will allow short us sleeps that I expect is what will be enough. Provided the platform supports it as not all do - in which case you will sleep for one tick (which can be 1ms or even more, depending on configuration). This is why I mentioned busy wait - for platforms that have long ticks it might be better to cycle for few us instead of sleeping. It assumes the retry delay is very short (in us).